### PR TITLE
chore: add categorical package code (ui-react-geo, ui-react-core-notifications and ui-react-notifications)

### PR DESCRIPTION
**Pull request by sreeramsama**
#### Description of changes
This PR reverts the commit 63cbba964aa2767791cba33260d537308e65c86f to add back the categorical packages - `ui-react-geo`, `ui-react-core-notifications` and `ui-react-notifications`. Components/Utilities from these packages will be migrated out of `ui-react` in a future breaking change release.

#### Description of how you validated changes
- `ui-react-geo`
  - unit and e2e tests
  - install and test features in sample apps (NextJS & React app with Vite)
- `ui-react-notifications`
  - unit and e2e tests
  - install and test features in sample apps (NextJS & React app with Vite)
- `ui-react-core-notifications` (used by `ui-react-native`)
  - unit and e2e tests
  - install and test features in sample mobile apps (Expo & RN CLI)

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .changeset/lemon-pumas-guess.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
```
Comment (calebpollman) [inline .changeset/lemon-pumas-guess.md:1]: Let's combine these three changesets in to one

File: examples/next/pages/ui/components/geo/map-with-location-search/index.page.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
-import { MapView, LocationSearch } from '@aws-amplify/ui-react';
 import { Amplify } from 'aws-amplify';
+import { MapView, LocationSearch } from '@aws-amplify/ui-react-geo';
 
 import '@aws-amplify/ui-react/styles.css';
```
[the remark this answers is not in the document]
Comment (sreeramsama) [reply examples/next/pages/ui/components/geo/map-with-location-search/index.page.tsx:4]: I kept the style export fix in a separate PR (https://github.com/aws-amplify/amplify-ui/pull/3913). Figured it is easier to review that way.
This PR mainly just reverts the revert of last week (https://github.com/aws-amplify/amplify-ui/pull/3884/).
I can add those changes here if that is preferred.

[the remark this answers is not in the document]
Comment (sreeramsama) [reply examples/next/pages/ui/components/geo/map-with-location-search/index.page.tsx]: Resolving comment as I've included the style related fixes in this PR

**Review by sreeramsama — COMMENTED** [review]
Things that need review:
- changeset
- GitHub workflow changes
- package.json files to make sure we only use dependencies we need and nothing more.
- style exports from each package

Other than that, the major portion of this PR is moving files from ui-react to the new package directories, updating import statements and adding deprecation messages for these components in ui-react where necessary. Will add specific comments on other things that have changed when moving.

**Review by sreeramsama — COMMENTED** [review]
Need advice on new README files we have for these new packages. For now, they are the same as we have for other new packages like react-storage and react-liveness. Can we change this template now to describe the specific components these packages offer?

File: packages/react-notifications/src/components/InAppMessaging/BannerMessage/__tests__/BannerMessage.test.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    return acc;
+  },
+  []
+);
+
+const TEST_PROPS: BannerMessageProps = { layout: 'TOP_BANNER' };
+
+const mockUseMessageProps = useMessageProps as jest.Mock;
+
+const useBreakpointValueSpy = jest.spyOn(UiReactModule, 'useBreakpointValue');
```
Comment (sreeramsama) [inline packages/react-notifications/src/components/InAppMessaging/BannerMessage/__tests__/BannerMessage.test.tsx:34]: Earlier, when we had InApp code in `ui-react`, we just imported `useBreakpointValue` from a relative path and mocked it. Changed it to now import it from `ui-react` and set a spy instead.

Same for other tests in `ui-react-notifications`.

**Comment by wlee221** [comment]
> Need advice on new README files we have for these new packages. For now, they are the same as we have for other new packages like react-storage and react-liveness. Can we change this template now to describe the specific components these packages offer?

Let's keep them as-is for now to be consistent with other categorical packages, and change all the packages together as a separate task.

**Review by wlee221 — APPROVED** [review]
LGTM

**Review by zchenwei — APPROVED** [review]
LGTM

File: packages/react/package.json [hunk 4]
```diff
@@ -49,6 +49,7 @@
   "dependencies": {
     "@aws-amplify/ui": "5.6.1",
     "@aws-amplify/ui-react-core": "2.1.20",
+    "@aws-amplify/ui-react-core-notifications": "0.0.1",
```
[the remark this answers is not in the document]
Comment (sreeramsama) [reply packages/react/package.json:52]: yes

**Review by ioanabrooks — APPROVED** [review]

**Pull request merged**